src/plotly/chartDefinition.ts

```diff
@@ -56,17 +55,31 @@ const getChartData = (
     );
   };
 
-  // Initialise a totals array to contain the totals
-  const seriesLength = chartData?.yValues[0].values.length;
-  let totals = new Array(seriesLength).fill(0);
-
+  const isAStackedBar = chartType === "stacked bar";
+  const xValues = chartData?.xValues;
   const allYSeries = chartData?.yValues;
 
+  // Get the unique X values
+  const allXValues = xValues.map((obj: any) => obj.values);
+  const uniqueXValues = Array.from(new Set(allXValues.flat()));
+
+  // Initialise an array to hold the cross-series totals for each point on the category axis
+  let totals = new Array(uniqueXValues.length).fill(0);
+
   // Iterate the available series and create a trace for each
-  allYSeries.map((series: any, index: number) => {
-    // Calculate the Y value totals across all series
-    for (let i = 0; i < series.values.length; i++) {
-      totals[i] += parseFloat(series.values[i]);
+  allYSeries.map((series: any, seriesIndex: number) => {
+    // If it's a stacked bar chart then calculate the cross-series totals
+    if (isAStackedBar) {
+      // Iterate through each of the unique X values (non sparse X values)
+      for (let i = 0; i < uniqueXValues.length; i++) {
+        const xValue = uniqueXValues[i];
+        const yValue = series.values[i];
+
+        // Update the total for this point on the category axis with the Y value if there's a sparse X match
+        if (xValues[seriesIndex].values.includes(xValue)) {
+          totals[i] += parseFloat(yValue);
```

Comment:
Noticed some `NaN`s appearing in the sparse scenario when we reviewed together.

Reply:
Fixed up the `NaN`s by tweaking the way around that the sparse and non sparse X value arrays are traversed. We now traverse the sparse array, store the X and Y values at each index and for each X value find the corresponding indexOf that X value in the non sparse array. This gives us the correct locations in the non sparse totals array to update with the Y value that corresponds to the sparse X value.